### DISCUSSION TITLE
Can't get plugin to successfully ping Samsung TV

### DISCUSSION THREAD

**Discussion by gradyhouston** [AlexGustafsson/homebridge-wol#146] [Questions]
### Backstory
My Samsung TVs (on an IoT network) go into this "deep sleep" after being off for 5-10 minutes. When this happens, for some reason, they turn off their connections and I can no longer turn them back on via my SmartThings/Homebridge/Homekit automations (which are on the private LAN).

### Issue
I installed homebridge-wol, the logs say it's pinging the IP address I supplied, but it only reports "offline". It doesn't appear to be 'waking' the device. The switch is created in homebridge and in homekit, but it doesn't do anything. 

### Details

This is the config:

```        {
            "name": "Basement Gaming TV",
            "ip": "192.168.3.16",
            "pingInterval": 2,
            "pingsToChange": 5,
            "pingTimeout": 1,
            "pingCommandTimeout": 0,
            "mac": "xxx",
            "broadcastAddress": "255.255.255.0",
            "startCommandTimeout": 0,
            "wakeGraceTime": 10,
            "wakeCommandTimeout": 0,
            "shutdownGraceTime": 15,
            "shutdownCommandTimeout": 0,
            "log": true,
            "logPinger": false,
            "debugLog": false,
            "returnEarly": false,
            "accessory": "NetworkDevice"
```

This is the log output:

`[1/28/2021, 14:13:52] [Basement Gaming TV] NetworkDevice awake cycle started for "Basement Gaming TV" (192.168.3.16)
[1/28/2021, 14:13:52] [Basement Gaming TV] Attempting to wake up "Basement Gaming TV" (192.168.3.16)
[1/28/2021, 14:13:52] [Basement Gaming TV] NetworkDevice "Basement Gaming TV" (192.168.3.16) went from status "Offline" to "Waking Up"
[1/28/2021, 14:13:52] [Basement Gaming TV] Attempting to start "Basement Gaming TV" (192.168.3.16) by sending a WoL packet
[1/28/2021, 14:13:52] [Basement Gaming TV] Waiting for wake grace time (30000ms) for "Basement Gaming TV" (192.168.3.16)`

And then it results with this after a few seconds:

`[1/28/2021, 14:14:25] [Basement Gaming TV] NetworkDevice "Basement Gaming TV" (192.168.3.16) went from status "Waking Up" to "Offline"`

### The need

Just help seeing if I've done something wrong or set up the plugin incorrectly.

**Comment by AlexGustafsson** [comment]
Have you been able to test that you can wake your TV up using WoL with another tool?

If I remember correctly WoL didn't work on a Samsung TV I tested - it used Bluetooth LE to wake up instead.

**Comment by gradyhouston** [comment]
I haven’t been able to find any other tools like this one, really.


On Thu, Jan 28, 2021 at 3:53 PM Alex Gustafsson <notifications@github.com>
wrote:

> Have you been able to test that you can wake your TV up using WoL with
> another tool?
>
> If I remember correctly WoL didn't work on a Samsung TV I tested - it used
> Bluetooth LE to wake up instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexGustafsson/homebridge-wol/issues/137#issuecomment-769383831>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH5FP6D3CR3J2MQSFGF6NGLS4HFDDANCNFSM4WXSWH6Q>
> .
>
-- 
Grady Houston

**Comment by AlexGustafsson** [comment]
Some routers have network tools including WoL. On macOS and Linux there's a package you can install. On Windows there's utilities as well.

Linux:
https://www.cyberciti.biz/faq/ubuntu-linux-wake-on-lan-client-command-installation-examples/

macOS:
https://apple.stackexchange.com/a/229596

Windows:
https://serverfault.com/questions/517/wake-on-lan-tool-for-windows

If no other tool works, then there's likely a larger issue. I did some quick research and it does indeed seem like Samsung TVs do not use Wake On LAN anymore. Someone speculated that this is due to them achieving a higher energy rating without it.

**Comment by badger13** [comment]
I have a Roku TV that goes into deep sleep 10-15 minutes after being turned off.  "Fast TV Start" Roku feature on that TV has issues with the screen, so I cannot keep it enabled, so that's why I'm looking at WOL.  Sometimes it works and will wake/turn on the TV, but sometime it doesn't work and just stays offline.  Any ideas why that might be the case?


For broadcast address, this is from Roku External Control Protocol page:
> Roku devices advertise their external control services using the multicast SSDP so that programs can discover the IP address of Roku devices in the area. There is a standard SSDP multicast address and port (239.255.255.250:1900) used for local area network communication. The Roku device responds to M-SEARCH queries on this IP address and port.


This is the config:
```            "name": "Basement Roku",
            "pingInterval": 2,
            "pingsToChange": 5,
            "pingTimeout": 1,
            "pingCommandTimeout": 0,
            "startCommandTimeout": 0,
            "wakeGraceTime": 20,
            "wakeCommandTimeout": 0,
            "shutdownGraceTime": 15,
            "shutdownCommandTimeout": 0,
            "log": true,
            "logPinger": false,
            "debugLog": false,
            "returnEarly": false,
            "ip": "192.168.xx.xxx",
            "mac": "xxx",
            "broadcastAddress": "239.255.255.250",
            "shutdownCommand": "curl -d ' ' \"http://192.168.xx.xxx:8060/keypress/PowerOff\""

**Comment by AlexGustafsson** [comment]
I'm going to go ahead and transfer this into a discussion. It does not seem to be an issue with the plugin, rather how many TVs go into a deep sleep and do not fully support WoL.
